systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaResource.java

```diff
@@ -234,29 +230,17 @@ private static Kafka getKafkaFromYaml(String yamlPath) {
      * Wait until the ZK, Kafka and EO are all ready
      */
     private static Kafka waitFor(Kafka kafka) {
-        String kafkaCrName = kafka.getMetadata().getName();
-        String namespace = kafka.getMetadata().getNamespace();
+        long timeout = Constants.TIMEOUT_FOR_RESOURCE_READINESS;
 
-        LOGGER.info("Waiting for Kafka {} in namespace {}", kafkaCrName, namespace);
-
-        StatefulSetUtils.waitForAllStatefulSetPodsReady(KafkaResources.zookeeperStatefulSetName(kafkaCrName), kafka.getSpec().getZookeeper().getReplicas());
-        StatefulSetUtils.waitForAllStatefulSetPodsReady(KafkaResources.kafkaStatefulSetName(kafkaCrName), kafka.getSpec().getKafka().getReplicas());
-
-        // EO should not be deployed if it does not contain UO and TO
-        if (kafka.getSpec().getEntityOperator().getTopicOperator() != null || kafka.getSpec().getEntityOperator().getUserOperator() != null) {
-            DeploymentUtils.waitForDeploymentReady(KafkaResources.entityOperatorDeploymentName(kafkaCrName));
-        }
         // Kafka Exporter is not setup every time
         if (kafka.getSpec().getKafkaExporter() != null) {
-            DeploymentUtils.waitForDeploymentReady(KafkaExporterResources.deploymentName(kafkaCrName));
+            timeout += Constants.TIMEOUT_FOR_RESOURCE_CREATION;
         }
         // Cruise Control is not setup every time
         if (kafka.getSpec().getCruiseControl() != null) {
-            LOGGER.info("Waiting for Cruise Control pods");
-            DeploymentUtils.waitForDeploymentReady(CruiseControlResources.deploymentName(kafkaCrName));
-            LOGGER.info("Cruise Control pods are ready");
+            timeout += Constants.TIMEOUT_FOR_RESOURCE_CREATION;
         }
```

Comment:
I wonder if we really need this. In general, the operator would be waiting the operation timeout several times:
* for Zoo pods
* for Kafka pods
* for EO pod
* for KE pod
* for CC pod

In the code you seem to take the 3 as one block valid for single `TIMEOUT_FOR_RESOURCE_READINESS`. And then for some reason you think the KE and CC pods are worth each `TIMEOUT_FOR_RESOURCE_READINESS` again. 

If you think it is needed I'm fine with it. But it seems a bit weird why should KE and CC get this special treatment.

Reply:
I have there `TIMEOUT_FOR_RESOURCE_CREATION` and that's like 5 minutes. So the whole Kafka CR -> ZK, Kafka, EO - will have 14 minutes to be ready. If we set up KE or CC it will get +5minutes to whole timeout ... how I mentioned in description, it's only prerequisite and I'll change to proper time (3-5 minutes) in next PR, don't want to make a mess in our `Constants`.

Is it OK this way?

Reply:
And I hope that KE and CC will get up sooner than in 5 minutes. But, we know our race conditions.

Reply:
Ok, I guess that makes more sense.